### PR TITLE
Matlab support

### PR DESCRIPTION
**Pull request by craig-willis**
This PR 

* Adds `licenses` directory mounted into running containers
* Adds `--build-arg` to pass Matlab installation key during build

See https://github.com/whole-tale/deploy-dev/pull/39 for full test case.

### CODE REVIEW

File: gwvolman/utils.py [hunk 1]
```diff
@@ -253,6 +253,11 @@ def _launch_container(volumeName, nodeId, container_config, tale_id='', instance
         )
     host = 'tmp-{}'.format(new_user(12).lower())
 
+    # Add licences mount for STATA and Matlab support
+    mounts.append(
+            docker.types.Mount(type='bind', source="/licenses/", target="/licenses")
+    )
+
```
Comment (Xarthisius) [inline gwvolman/utils.py:260]: It needs to be umounted in `gwvolman/tasks.py:remove_volume`

**Comment by Xarthisius** [comment]
I'd like to suggest the following change:

```diff
--- a/gwvolman/utils.py
+++ b/gwvolman/utils.py
@@ -323,7 +323,10 @@ def _build_image(cli, tale_id, image, tag, temp_dir, repo2docker_version):
     container = cli.containers.run(
         image=repo2docker_version,
         command=r2d_cmd,
-        environment=['DOCKER_HOST=unix:///var/run/docker.sock'],
+        environment=[
+            "DOCKER_HOST=unix:///var/run/docker.sock",
+            "BUILDKIT_PROGRESS=plain",
+        ],
         privileged=True,
         detach=True,
         remove=True,
```

IMHO it makes the build log more useful for us.

**Comment by craig-willis** [comment]
@Xarthisius This is already done in my `repo2docker` PR via the `--progress plain` flag.

https://github.com/whole-tale/repo2docker/blob/matlab/repo2docker/docker_utils.py#L25

File: gwvolman/utils.py (hunk 1 above)
Comment (craig-willis) [reply gwvolman/utils.py:260]: I've added the umount call to this and the Stata PR https://github.com/whole-tale/gwvolman/pull/120/commits/89b322baf12b9287d0d439ad38f32e15304ace70

Comment (craig-willis) [reply gwvolman/utils.py:260]: As discussed, this is not needed

**Review by Xarthisius — APPROVED** [review]

**Pull request merged**